### ISSUE TITLE
about the training

### ISSUE THREAD

**Issue by HuShaohanAI** [ClementPinard/FlowNetPytorch#38]
Hi~
I have training my model with the same config as yours. But, the decay_loss decrease too faster, and after about 6000 iterations, the loss become nan.
So I want know your training configurations, are they the default configurations in your main.py?

**Comment by ClementPinard** [comment]
Hi,
the default configuration should lead to convergence
As for my pretrained networks, you can see them here : https://github.com/ClementPinard/FlowNetPytorch#training-results  All other parameters as same as default

What do you mean by decay_loss ? Are your talking about learning rate decay or about downscaling loss weights? Are you training on a custom dataset or one of the three ones in the repo ?

**Comment by HuShaohanAI** [comment]
the decay loss is the L2 regularization loss term, it is the square of the weight parameters. But now I have already fix that problem.
But another is that if I use your default config parameters, after traning 300 epochs, the mepe on the val set is more than 5, I don't know what's the wrong. I just use the Offical FlyingChairs dataset that has 22872 pairs in total. And I also do the same data augmentations as yours in flow_transforms.py.
I have tried to decrease the weight decay from your 4e-4 to a smaller one 1e-5, then, after more than 400 epochs, the mepe on the val set is 2.78. But your mepe is 1.951

**Comment by jangho2001us** [comment]
@HuShaohanAI 
Hi, I suffered from a similar problem with you (error comes to nan).
I think training optical flow is very sensitive to a learning rate :)

**Comment by ClementPinard** [comment]
Weight decay is most likely to be rather optional if your only concern is to perform validation on flying chair. However you might not get better results on other validation dataset (without finetuning) since it will have learned too hard to recognize the dataset I believe.

As for my validation error, I had set up a very long training schedule, note the `epoch_size` which is far greater than default values.

I guess your main problem is the learning rate schedule. Namely, we divide it by 2 at epochs 100, 150 and 200. So you might want to either
 - do many epochs, but change the milestones (say, 200, 250 and 300, note that the number is not fixed, you can have 2 milestones, or 4 or whatever)
 - or do large epochs. Keep in mind that an epoch cannot be above 1 full dataset, which is `2700` for a batchsize of `8`. `epoch_size` is given in number of iterations. It means that for a batchsize of `16`, the max epoch_size will be `1350`

**Comment by HuShaohanAI** [comment]
My epoch_size is 2700, and my schedule is [100, 150, 200], and if I use 4e-4 weight decay, after 300 epochs ,the mepe on val set is 5.4.  I could get 2.77 on val set if I use 1e-5 weight decay.
Should I increase the milestone to [200, 250, 300] or [250, 300, 350]?

**Comment by ClementPinard** [comment]
Hmm that's indeed odd then. Maybe something has changed in torch 0.4 since 0.3. I'll try to make a new training to see if I still get these results

**Comment by HuShaohanAI** [comment]
Well I didn't use the pytorch framework, I used our own deeplearning framework in my company instead. But I have used the same configuration parameters as yours, but the mepe on val is far from yours. (5.4   vs   1.951)

**Comment by ClementPinard** [comment]
In that case, I guess there's a possibility there is a bug somewhere in your implementation.
Do you apply the same data augmentation ?
Do you divide optical flow by 20 ?

If you have the possibility to test it on pytorch, I think you should run both to compare the training thoroughly.

**Comment by HuShaohanAI** [comment]
Thanks for your answer!
I will check my code carefully again.
